### PR TITLE
update signalfx plugin to work on macosx

### PR DESCRIPTION
**Pull request by dgriff-tools**
Theses changes were made to support running this plugin on macosx.  The primary differences were between linux /proc data and the equivalent on macosx.  Also the existing get network interface method only supported linux so macosx only will use netifaces.

### CODE REVIEW

File: src/signalfx_metadata.py [hunk 1]
```diff
@@ -849,32 +875,49 @@ def get_interfaces(host_info={}):
...
+
+        num_processor_result = popen(["system_profiler", "SPHardwareDataType"])
+        for x in num_processor_result.splitlines():
+            if x.strip().startswith("Number of Processors"):
+                host_info["host_physical_cpus"] = \
+                    x.strip().split(":")[1].strip()
+                break
+
+    else:
+        """populate host_info with cpu information"""
```
Comment (mdubbyap) [inline src/signalfx_metadata.py]: nit: this was method pydoc, not specific to non darwin

File: src/signalfx_metadata.py [hunk 2]
```diff
@@ -801,40 +816,51 @@ def send():
...
-        if outbytes == _bytes:
-            max_possible *= 2
-        else:
-            break
-    namestr = names.tostring()
-    ifaces = []
-    for i in range(0, outbytes, struct_size):
-        iface_name = bytes.decode(namestr[i:i + 16]).split('\0', 1)[0]
-        iface_addr = socket.inet_ntoa(namestr[i + 20:i + 24])
-        ifaces.append((iface_name, iface_addr))
```
Comment (mdubbyap) [inline src/signalfx_metadata.py:835]: netifaces can be used for linux as well, why not use it for both?

File: src/signalfx_metadata.py [hunk 3]
```diff
@@ -801,40 +816,51 @@ def send():
 
 
 def all_interfaces():
-    """
```
Comment (mdubbyap) [inline src/signalfx_metadata.py:804]: want some method pydoc

File: src/signalfx_metadata.py [hunk 4]
```diff
@@ -1107,19 +1159,34 @@ def send_top():
...
+                    get_priority(p.pid),  # priority
+                    p.nice(),  # nice value, numerical
+                    p.memory_info_ex()[1] / 1024,  # virtual memory size in kb
+                    p.memory_info_ex()[0] / 1024,  # resident memory size in kb
+                    p.memory_info_ex()[2] / 1024,  # shared memory size in kb
+                    status_map.get(p.status(), "D"),  # process status
+                    p.cpu_percent(),  # % cpu, float
+                    p.memory_percent(),  # % mem, float
+                    to_time(p.cpu_times().system + p.cpu_times().user),  # cpu
+                    p.name()  # command
```
Comment (mdubbyap) [inline src/signalfx_metadata.py]: These look really close, like more than 90%. Why not have get_command do the if darwin bs  so they can be the same?

File: src/signalfx_metadata.py [hunk 5]
```diff
@@ -1134,11 +1201,16 @@ def send_top():
...
 def get_memory(host_info):
-    """get total physical memory for machine"""
-    with open("/proc/meminfo") as f:
-        pieces = f.readline()
-        _, mem_total, _ = pieces.split()
-        host_info["host_mem_total"] = mem_total
+
+    if sys.platform == 'darwin':
+        host_info["host_mem_total"] = \
+            str(int(popen(["sysctl", "-n", "hw.memsize"])) / 1024)
```
Comment (mdubbyap) [inline src/signalfx_metadata.py:1203]: are the units the same between these two?

File: src/signalfx_metadata.py (hunk 1 above)
Comment (dgriff-tools) [inline src/signalfx_metadata.py]: fixing that now

File: src/signalfx_metadata.py (hunk 2 above)
Comment (dgriff-tools) [inline src/signalfx_metadata.py:835]: I didn't want to bring in the bloat/dependency of netifaces for linux just to support macosx.

File: src/signalfx_metadata.py (hunk 3 above)
Comment (dgriff-tools) [inline src/signalfx_metadata.py:804]: done

File: src/signalfx_metadata.py (hunk 5 above)
Comment (dgriff-tools) [inline src/signalfx_metadata.py:1203]: yes. validated

File: src/signalfx_metadata.py (hunk 4 above)
Comment (dgriff-tools) [inline src/signalfx_metadata.py]: it is more like ~80% the same (2 out of 11 different).  But I down with the reason so on the way.

**Comment by mdubbyap** [comment]
You'll want to create a corresponding change to collectd-build-\* to rev the version and merge both at the same time (or the version first).  How is the packaging done for mac?

**Comment by dgriff-tools** [comment]
there is a new repo named collectd-build-osxpkg that is modeled like the ubuntu and rpm repos but builds to a macosx pkg instead. I have the build-collectd mostly done but need to make the build-plugin part (which needs this merge here).  Once I have push the collectd-build-osxpkg repo I will look at eiter changing collectd-install to support osx pkgs/os or have the pkg use/call the collectd-install bash functions.  This work will complete this week (so I can rev the versions for all collectd build repos then I guess).

**Comment by mdubbyap** [comment]
But if you merge this first, you will replace the existing test repo's version with this one, which is really the next version.  So you still need to create the build-collectd\* PRs first.

**Comment by dgriff-tools** [comment]
ok.  I will have that PR later this week and we can revisit this merge then. And then the other build repos for a version rev.

**Pull request merged**